Common/Mapper/TypeMapper.cs

```diff
@@ -41,6 +42,7 @@ private T typeConverter<T>(dynamic dynamicObj)
             if (typeof(T) == typeof(DeviceND))
             {
                 FixIsSimulatedDevice(dynamicObj);
+                DeviceSchemaHelper.FixDeviceSchema(dynamicObj);
```

Comment:
I am fixing the dynamic object here. ND version will not have this function at all